### PR TITLE
Util: a optimization of mysql.Decimal in Datum

### PR DESCRIPTION
**Pull request by zxylvlp**
The space of Datum to save mysql.Decimal can be saved by just a ptr.

### CODE REVIEW

File: util/types/datum.go [hunk 1]
```diff
@@ -315,8 +315,10 @@ func (d *Datum) SetValue(val interface{}) {
 		d.SetBytes(x)
 	case mysql.Bit:
 		d.SetMysqlBit(x)
-	case mysql.Decimal:
+	case *mysql.Decimal:
 		d.SetMysqlDecimal(x)
+	case mysql.Decimal:
```
Comment (shenli) [inline util/types/datum.go]: Can we remove this case?

Comment (zxylvlp) [inline util/types/datum.go]: We still have some setValue statements to put decimal in datum. But this case will be removed in the futher.

**Comment by shenli** [comment]
LGTM

**Comment by zimulala** [comment]
LGTM

**Pull request closed without merging**